### ISSUE TITLE
GANSynth error while training

### ISSUE THREAD

**Issue by georgeMuraveiAlkh** [tensorflow/magenta#1489]
Getting this 

<details>
<summary>error</summary>

```python
Traceback (most recent call last):
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 1334, in _do_call
    return fn(*args)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 1319, in _run_fn
    options, feed_dict, fetch_list, target_list, run_metadata)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 1407, in _call_tf_sessionrun
    run_metadata)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Computed output size would be negative: -14 [input_size: 1, effective_filter_size: 16, stride: 1]
	 [[{{node Generator/progressive_gan_generator/progressive_gan_block_1/conv0/conv2d/Conv2D}}]]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/anaconda3/bin/gansynth_train", line 11, in <module>
    sys.exit(console_entry_point())
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/gansynth_train.py", line 136, in console_entry_point
    tf.app.run(main)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/platform/app.py", line 125, in run
    _sys.exit(main(argv))
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/gansynth_train.py", line 132, in main
    run(flags)
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/gansynth_train.py", line 101, in run
    train_util.train(model, **config)
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/lib/train_util.py", line 516, in train
    save_summaries_steps=(kwargs['save_summaries_num_images']))
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/contrib/gan/python/train.py", line 1198, in gan_train
    config=config)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/contrib/training/python/training/training.py", line 546, in train
    loss = session.run(train_op, run_metadata=run_metadata)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/training/monitored_session.py", line 676, in run
    run_metadata=run_metadata)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/training/monitored_session.py", line 1171, in run
    run_metadata=run_metadata)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/training/monitored_session.py", line 1270, in run
    raise six.reraise(*original_exc_info)
  File "/anaconda3/lib/python3.6/site-packages/six.py", line 693, in reraise
    raise value
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/training/monitored_session.py", line 1255, in run
    return self._sess.run(*args, **kwargs)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/training/monitored_session.py", line 1319, in run
    feed_dict, options)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/training/monitored_session.py", line 1345, in _call_hook_before_run
    request = hook.before_run(run_context)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/contrib/gan/python/train.py", line 1059, in before_run
    run_context.session.run(self._train_ops)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 929, in run
    run_metadata_ptr)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 1152, in _run
    feed_dict_tensor, options, run_metadata)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 1328, in _do_run
    run_metadata)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 1348, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Computed output size would be negative: -14 [input_size: 1, effective_filter_size: 16, stride: 1]
	 [[node Generator/progressive_gan_generator/progressive_gan_block_1/conv0/conv2d/Conv2D (defined at /anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/lib/layers.py:287) ]]

Caused by op 'Generator/progressive_gan_generator/progressive_gan_block_1/conv0/conv2d/Conv2D', defined at:
  File "/anaconda3/bin/gansynth_train", line 11, in <module>
    sys.exit(console_entry_point())
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/gansynth_train.py", line 136, in console_entry_point
    tf.app.run(main)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/platform/app.py", line 125, in run
    _sys.exit(main(argv))
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/gansynth_train.py", line 132, in main
    run(flags)
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/gansynth_train.py", line 95, in run
    model = lib_model.Model(stage_id, batch_size, config)
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/lib/model.py", line 264, in __init__
    generator_inputs=(noises, gen_one_hot_labels))
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/contrib/gan/python/train.py", line 112, in gan_model
    generated_data = generator_fn(generator_inputs)
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/lib/model.py", line 261, in <lambda>
    generator_fn=lambda inputs: g_fn(inputs)[0],
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/lib/model.py", line 256, in <lambda>
    g_fn = lambda x: net_fns.g_fn_registry[config['g_fn']](x, **config)
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/lib/network_functions.py", line 52, in generator_fn_specgram
    simple_arch=kwargs['simple_arch'])
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/lib/networks.py", line 372, in generator
    x = _conv2d('conv0', x, (start_h, start_w), num_filters_fn(1), 'VALID')
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/lib/networks.py", line 338, in _conv2d
    scope=scope)
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/lib/layers.py", line 296, in custom_conv2d
    use_weight_scaling=use_weight_scaling)
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/lib/layers.py", line 235, in _custom_layer_impl
    output = post_scale * apply_kernel(kernel_shape, kernel_initializer) + bias
  File "/anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/lib/layers.py", line 287, in _apply_kernel
    kernel_initializer=kernel_initializer)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/util/deprecation.py", line 324, in new_func
    return func(*args, **kwargs)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/layers/convolutional.py", line 424, in conv2d
    return layer.apply(inputs)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/keras/engine/base_layer.py", line 1227, in apply
    return self.__call__(inputs, *args, **kwargs)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/layers/base.py", line 530, in __call__
    outputs = super(Layer, self).__call__(inputs, *args, **kwargs)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/keras/engine/base_layer.py", line 554, in __call__
    outputs = self.call(inputs, *args, **kwargs)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/keras/layers/convolutional.py", line 194, in call
    outputs = self._convolution_op(inputs, self.kernel)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/nn_ops.py", line 966, in __call__
    return self.conv_op(inp, filter)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/nn_ops.py", line 591, in __call__
    return self.call(inp, filter)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/nn_ops.py", line 208, in __call__
    name=self.name)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/gen_nn_ops.py", line 1026, in conv2d
    data_format=data_format, dilations=dilations, name=name)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 788, in _apply_op_helper
    op_def=op_def)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 3300, in create_op
    op_def=op_def)
  File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1801, in __init__
    self._traceback = tf_stack.extract_stack()

InvalidArgumentError (see above for traceback): Computed output size would be negative: -14 [input_size: 1, effective_filter_size: 16, stride: 1]
	 [[node Generator/progressive_gan_generator/progressive_gan_block_1/conv0/conv2d/Conv2D (defined at /anaconda3/lib/python3.6/site-packages/magenta/models/gansynth/lib/layers.py:287) ]]
```
</details>

 while training on ```nsynth-test.tfrecord``` with 
```console
gansynth_train --config=mel_prog_hires --hparams='{"train_data_path":"nsynth-train.tfrecord", "train_root_dir":"/tmp/gansynth/train"}'
```
If not use ```--config=mel_prog_hires``` config error became:
```python
InvalidArgumentError (see above for traceback): Computed output size would be negative: -2 [input_size: 1, effective_filter_size: 4, stride: 1]
         [[node Generator/progressive_gan_generator/progressive_gan_block_1/conv0/conv2d/Conv2D (defined at /Users/georgezagraid/Projects/AI/Projects/GAN/gansynth/lib/layers.py:287) ]]
```
Please help.

**Comment by jesseengel** [comment]
Hi George, 

Sorry you're having trouble. I'm not able to reproduce your error on my local linux box. When I run

```
gansynth_train --config=mel_prog_hires --hparams='{"train_data_path":"~/nsynth-test.tfrecord", "train_root_dir":"/tmp/gansynth/train"}'
```

It seems to train fine for me. Have you updated to the latest pip package?

**Comment by georgeMuraveiAlkh** [comment]
@jesseengel thanx for reply. I've got version which pip installed. My env:
```
MacOS Mojave 10.14.4
pip 19.0.3
Python 3.6.5 Anaconda
tensorflow==1.13.1
numpy==1.16.2
magenta==1.0.5
```
<details>
<summary>All intalled packages list</summary>

```
absl-py==0.4.0
alabaster==0.7.10
anaconda-client==1.6.14
anaconda-navigator==1.9.6
anaconda-project==0.8.2
appnope==0.1.0
appscript==1.0.1
asn1crypto==0.24.0
astor==0.7.1
astroid==1.6.3
astropy==3.0.2
attrs==18.1.0
audioread==2.1.6
autobahn==18.9.2
Automat==0.7.0
Babel==2.5.3
backcall==0.1.0
backports.shutil-get-terminal-size==1.0.0
backports.tempfile==1.0
backports.weakref==1.0.post1
beautifulsoup4==4.6.0
bitarray==0.8.1
bkcharts==0.2
blaze==0.11.3
bleach==2.1.3
bokeh==0.12.16
boto==2.48.0
Bottleneck==1.2.1
bz2file==0.98
cachetools==3.1.0
certifi==2018.4.16
cffi==1.11.5
chardet==3.0.4
click==6.7
cloudpickle==0.5.3
clyent==1.2.2
colorama==0.3.9
conda==4.6.8
conda-build==3.10.5
conda-verify==2.0.0
constantly==15.1.0
contextlib2==0.5.5
cryptography==2.2.2
cycler==0.10.0
Cython==0.28.2
cytoolz==0.9.0.1
dask==0.17.5
datashape==0.5.4
dateparser==0.7.0
decorator==4.3.0
distributed==1.21.8
docutils==0.14
dopamine-rl==2.0.1
entrypoints==0.2.3
enum34==1.1.6
et-xmlfile==1.0.1
fastcache==1.0.2
filelock==3.0.4
Flask==1.0.2
Flask-Cors==3.0.4
future==0.17.1
gast==0.2.0
gevent==1.3.0
gin-config==0.1.4
glob2==0.6
gmpy2==2.0.8
google-api-python-client==1.7.8
google-auth==1.6.3
google-auth-httplib2==0.0.3
googleapis-common-protos==1.5.9
greenlet==0.4.13
grpcio==1.12.1
gunicorn==19.9.0
gym==0.12.1
h5py==2.7.1
heapdict==1.0.0
html5lib==1.0.1
httplib2==0.12.1
hyperlink==18.0.0
idna==2.6
imageio==2.3.0
imagesize==1.0.0
incremental==17.5.0
int-date==0.1.8
intervaltree==3.0.2
ipykernel==4.8.2
ipython==6.4.0
ipython-genutils==0.2.0
ipywidgets==7.2.1
isort==4.3.4
itsdangerous==0.24
jax==0.1.22
jaxlib==0.1.12
jdcal==1.4
jedi==0.12.0
Jinja2==2.10
joblib==0.13.2
jsonschema==2.6.0
jupyter==1.0.0
jupyter-client==5.2.3
jupyter-console==5.2.0
jupyter-core==4.4.0
jupyterlab==0.32.1
jupyterlab-launcher==0.10.5
Keras==2.2.4
Keras-Applications==1.0.7
Keras-Preprocessing==1.0.9
kfac==0.1.4
kiwisolver==1.0.1
lazy-object-proxy==1.3.1
librosa==0.6.3
llvmlite==0.23.1
locket==0.2.0
lxml==4.2.1
magenta==1.0.5
Markdown==2.6.11
MarkupSafe==1.0
matplotlib==2.2.2
mccabe==0.6.1
mesh-tensorflow==0.0.5
mido==1.2.6
mir-eval==0.5
mistune==0.8.3
mkl-fft==1.0.0
mkl-random==1.0.1
mock==2.0.0
more-itertools==4.1.0
mpmath==1.0.0
msgpack==0.5.6
msgpack-python==0.5.6
multipledispatch==0.5.0
navigator-updater==0.2.1
nbconvert==5.3.1
nbformat==4.4.0
networkx==1.8.1
nltk==3.3
nose==1.3.7
notebook==5.5.0
numba==0.38.0
numexpr==2.6.5
numpy==1.16.2
numpydoc==0.8.0
oauth2client==4.1.3
odo==0.5.1
olefile==0.45.1
opencv-python==4.0.0.21
openpyxl==2.5.3
opt-einsum==2.3.2
packaging==17.1
pandas==0.23.0
pandocfilters==1.4.2
parso==0.2.0
partd==0.3.8
path.py==11.0.1
pathlib2==2.3.2
patsy==0.5.0
pbr==5.1.3
pep8==1.7.1
pexpect==4.5.0
pickleshare==0.7.4
Pillow==5.1.0
pkginfo==1.4.2
plotly==3.1.1
pluggy==0.6.0
ply==3.11
pretty-midi==0.2.8
promise==2.2.1
prompt-toolkit==1.0.15
protobuf==3.7.1
psutil==5.4.5
ptyprocess==0.5.2
py==1.5.3
pyasn1==0.4.4
pyasn1-modules==0.2.2
pycodestyle==2.4.0
pycosat==0.6.3
pycparser==2.18
pycrypto==2.6.1
pycurl==7.43.0.1
pyflakes==1.6.0
pyglet==1.3.2
Pygments==2.2.0
pygtrie==2.3
PyHamcrest==1.9.0
pylint==1.8.4
pyodbc==4.0.23
pyOpenSSL==18.0.0
pyparsing==2.2.0
pypng==0.0.19
PySocks==1.6.8
pytest==3.5.1
pytest-arraydiff==0.2
pytest-astropy==0.3.0
pytest-doctestplus==0.1.3
pytest-openfiles==0.3.0
pytest-remotedata==0.2.1
python-binance==0.7.0
python-dateutil==2.7.3
python-rtmidi==1.1.2
pytz==2018.4
PyWavelets==0.5.2
PyYAML==3.12
pyzmq==17.0.0
QtAwesome==0.4.4
qtconsole==4.3.1
QtPy==1.4.1
regex==2018.8.29
requests==2.18.4
resampy==0.2.1
retrying==1.3.3
rope==0.10.7
rsa==4.0
ruamel-yaml==0.15.35
scikit-image==0.13.1
scikit-learn==0.19.1
scipy==1.1.0
seaborn==0.8.1
Send2Trash==1.5.0
service-identity==17.0.0
simplegeneric==0.8.1
singledispatch==3.4.0.3
six==1.11.0
sk-video==1.1.10
snowballstemmer==1.2.1
sonnet==0.1.6
sortedcollections==0.6.1
sortedcontainers==2.1.0
sox==1.3.7
Sphinx==1.7.4
sphinxcontrib-websupport==1.0.1
spyder==3.2.8
SQLAlchemy==1.2.7
statsmodels==0.9.0
stevedore==1.30.0
stockstats==0.2.0
sympy==1.1.1
ta==0.2.0
tables==3.4.3
tblib==1.3.2
tensoflow==0.0.5
tensor2tensor==1.13.1
tensorboard==1.13.1
tensorflow==1.13.1
tensorflow-datasets==1.0.1
tensorflow-estimator==1.13.0
tensorflow-metadata==0.13.0
tensorflow-probability==0.6.0
termcolor==1.1.0
terminado==0.8.1
testpath==0.3.1
toolz==0.9.0
tornado==5.0.2
tqdm==4.31.1
traitlets==4.3.2
Twisted==18.7.0
txaio==18.8.1
typing==3.6.4
tzlocal==1.5.1
unicodecsv==0.14.1
uritemplate==3.0.0
urllib3==1.22
virtualenv==16.1.0
virtualenv-clone==0.4.0
virtualenvwrapper==4.8.2
wcwidth==0.1.7
webencodings==0.5.1
Werkzeug==0.14.1
widgetsnbextension==3.2.1
wrapt==1.10.11
xlrd==1.1.0
XlsxWriter==1.0.4
xlwings==0.11.8
xlwt==1.2.0
zict==0.1.3
zope.interface==4.5.0
```
</details>

I can provide any additional information you need. Is this exception could be a result of wrong versions of dependencies?

**Comment by jesseengel** [comment]
Hey, I just did a fresh install with miniconda2 on my MacBookPro (Mojave 10.14.3) following  `https://github.com/tensorflow/magenta/blob/master/magenta/tools/magenta-install.sh`, and it worked. Can you confirm if that works for you? You might also be able to get away with just setting up a new conda environment.

**Comment by georgeMuraveiAlkh** [comment]
@jesseengel thanx for you help. I tryed before: played with versions of dependencies - no results, create new conda env - no results. Finally I just finished total reinstall of Anaconda3 and it finally work! Of course I cant finish training on macbook but at least it started and I was able to generate sounds with pretrained models. So I assume that something was broken inside my anaconda (however everything else worked fine). Thanx again for your help.
